src/app/diary-module/containers/sign-in/sign-in.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'app-sign-in',
+  templateUrl: './sign-in.component.html',
+  styleUrls: ['./sign-in.component.scss'],
+})
+export class SignInComponent implements OnInit {
+  username = new FormControl('');
+  disable = new FormControl(true);
```

Comment:
can do this with single form control.
`username = new FormControl('',[Validators.required]);`
after this you can change the button's state with `username.valid`